include/boost/regex/v5/perl_matcher_common.hpp

```diff
@@ -60,12 +60,7 @@ void perl_matcher<BidiIterator, Allocator, traits>::construct_init(const basic_r
    if(e.empty())
    {
       // precondition failure: e is not a valid regex.
-      std::invalid_argument ex("Invalid regular expression object");
-#ifndef BOOST_REGEX_STANDALONE
-      boost::throw_exception(ex);
-#else
-      throw e;
-#endif
+      BOOST_REGEX_DETAIL_THROW("Invalid regular expression object");
```

Comment:
Ideally this should still throw a `std::invalid_argument`, otherwise we're breaking the interface?